toxics/timeout.go

```diff
@@ -12,19 +12,31 @@ type TimeoutToxic struct {
 func (t *TimeoutToxic) Pipe(stub *ToxicStub) {
 	timeout := time.Duration(t.Timeout) * time.Millisecond
 	if timeout > 0 {
+		for {
+			select {
+			case <-time.After(timeout):
+				stub.Close()
+				return
+			case <-stub.Interrupt:
+				return
+			case <-stub.Input:
+				// Drop the data on the ground.
+			}
+		}
+	} else {
```

Comment:
This is keeping consistent with the previous implementation of the `timeout` toxic. Where a `0` timeout value immediately stops all data.

I agree that the name `timeout` might be confusing. I would propose we open an issue unless you are confident in how this should be renamed / changed.